nuclio_sdk/event.py

```diff
@@ -88,35 +88,31 @@ def get_header(self, header_key):
 
     @staticmethod
     def from_msgpack(parsed_data):
-        """Decode msgpack event encoded as JSON by processor"""
-
-        # extract content type, needed to decode body
-        content_type = parsed_data["content_type"]
-        body = Event.decode_msgpack_body(parsed_data["body"], content_type)
-        return Event.from_parsed_data(parsed_data, body, content_type)
+        """
+        Deprecated. use nuclio_sdk.event.EventDeserializerFactory.create("msgpack").deserialize(parsed_data) instead.
+        To be removed on >= 0.4.0
+        """
+        return EventDeserializerFactory.create("msgpack").deserialize(parsed_data)
 
     @staticmethod
     def from_json(data):
-        """Decode event encoded as JSON by processor"""
-
-        parsed_data = json.loads(data)
-
-        # extract content type, needed to decode body
-        content_type = parsed_data["content_type"]
-        body = Event.decode_body(parsed_data["body"], content_type)
-        return Event.from_parsed_data(parsed_data, body, content_type)
+        """
+        Deprecated. use nuclio_sdk.event.EventDeserializerFactory.create("json").deserialize(data) instead.
+        To be removed on >= 0.4.0
+        """
+        return EventDeserializerFactory.create("json").deserialize(data)
 
     @classmethod
-    def from_parsed_data(cls, parsed_data, body, content_type):
+    def from_parsed_data(cls, parsed_data, body):
         trigger = TriggerInfo(
             parsed_data["trigger"]["kind"], parsed_data["trigger"]["name"]
         )
         return cls(
             body=body,
-            content_type=content_type,
+            content_type=parsed_data["content_type"],
             trigger=trigger,
-            fields=parsed_data.get("fields"),
-            headers=parsed_data.get("headers"),
+            fields=parsed_data["fields"],
+            headers=parsed_data["headers"],
```

Comment:
why did you replace the safer `get`s here? I assume there was a good reason for them (keys may be missing?)

Reply:
it was implemented a long time ago, my guess is that it was not in yet on the processor level while it was written here.
I double checked and processor _always_ sends these fields.

https://github.com/nuclio/nuclio/blob/development/pkg/processor/runtime/rpc/encode.go#L36-L37